### PR TITLE
WIP: Introducing VPC Flow Logs

### PR DESCRIPTION
**Pull request by ljakimczuk**
Hi! 

I'm new to `moto` :) I would like to add support for the VPC Flow Logs. I'm open to and suggestions, let me know if the approach I took fits the rest of the library.

### CODE REVIEW

File: moto/ec2/models.py [hunk 1]
```diff
@@ -3511,6 +3524,295 @@ def modify_subnet_attribute(self, subnet_id, attr_name, attr_value):
...
+        log_format,
+        max_aggregation_interval,
+    ):
+        # Guess it's best to put it here due to possible
+        # lack of them in the CloudFormation template
+        max_aggregation_interval = (
+            "600" if max_aggregation_interval is None else max_aggregation_interval
+        )
+        log_destination_type = (
+            "cloud-watch-logs" if log_destination_type is None else log_destination_type
```
Comment (bblommers) [inline moto/ec2/models.py:3702]: Can we have a test where LogDestinationType is explicitly set to `cloud-watch-logs`? At the moment there are only tests for s3, and tests where the param is not provided (None)

File: tests/test_ec2/test_flow_logs.py [hunk 2]
```diff
@@ -0,0 +1,621 @@
+from __future__ import unicode_literals
+
+# Ensure 'assert_raises' context manager support for Python 2.6
```
Comment (bblommers) [inline tests/test_ec2/test_flow_logs.py:3]: Moto does not support Python 2.6, so I don't think we need this

File: tests/test_ec2/test_flow_logs.py [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+import tests.backport_assert_raises  # noqa
+from nose.tools import assert_raises
+
+import boto3
+
+# boto3.set_stream_logger(name='botocore')
```
Comment (bblommers) [inline tests/test_ec2/test_flow_logs.py:9]: Not necessary

File: moto/ec2/models.py [hunk 4]
```diff
@@ -3511,6 +3524,295 @@ def modify_subnet_attribute(self, subnet_id, attr_name, attr_value):
...
+                self.get_vpc(resource_id)
+            elif resource_type == "Subnet":
+                # Validate Subnets exist
+                self.get_subnet(resource_id)
+            elif resource_type == "NetworkInterface":
+                # Validate NetworkInterfaces exist
+                self.get_network_interface(resource_id)
+
+            if log_destination_type == "s3":
+                arn = log_destination.split(":", 5)[5]
```
Comment (bblommers) [inline moto/ec2/models.py:3737]: It hasn't been released yet, but we're introducing a dependency model where the user would only have to install the dependencies for the specific service he's interested in. 
(See the top of the Readme for installation instructions on what that would look like.)

The new `moto.logs` and `moto.s3` imports at the top would mean that everyone who uses EC2, would also have to install dependencies for Logs and S3.
Can we move the import statements inside the if/elif statement, so the import only happens when we absolutely need it? 
That way, only users who use S3 as a destination type would have to install dependencies for S3.

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @ljakimczuk, thanks for your PR, and welcome to Moto! I left a few comments, but  the overall approach looks very good.

File: moto/ec2/models.py (hunk 1 above)
Comment (ljakimczuk) [reply moto/ec2/models.py:3702]: Sure, I'll add this

File: moto/ec2/models.py (hunk 4 above)
Comment (ljakimczuk) [reply moto/ec2/models.py:3737]: Haven't thought that, it sounds smart, let me change that and thanks for catching that! :)

**Review by bblommers — APPROVED** [review]
LGTM. Thanks again, @ljakimczuk!

**Comment by ljakimczuk** [comment]
@bblommers thank you, glad you liked that!

**Pull request merged**